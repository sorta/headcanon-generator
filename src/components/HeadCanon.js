import React, { Component } from 'react';
import PropTypes from 'prop-types';

class HeadCanon extends Component {
  static propTypes = {
    subject: PropTypes.shape({
      name: PropTypes.string,
      fandomKey: PropTypes.string,
    }),
    descriptor: PropTypes.shape({
      name: PropTypes.string,
    }),
    fandoms: PropTypes.object,
  };
  static defaultProps = {
    subject: {},
    descriptor: {},
    fandoms: {},
  };

  render() {
    if (Object.keys(this.props.subject).length === 0 || Object.keys(this.props.descriptor).length === 0) {
      return null;
    }

    let subjectSection = (
      <h2 className="HeadCanon-item HeadCanon-Subject">{this.props.subject.name}</h2>
    );

    if ({}.hasOwnProperty.call(this.props.fandoms, this.props.subject.fandomKey)) {
      subjectSection = (
        <h2 className="HeadCanon-item HeadCanon-Subject"
          data-tooltip={`From "${this.props.fandoms[this.props.subject.fandomKey].name}"`}
        >{this.props.subject.name}</h2>
      );
    }

    return (
      <div className="HeadCanon">
        {subjectSection}
        &nbsp;
        <h2 className="HeadCanon-item HeadCanon-Verb">is</h2>
        &nbsp;
        <h2 className="HeadCanon-item HeadCanon-Descriptor">{this.props.descriptor.name}</h2>
      </div>
    );
  }
}

export default HeadCanon;
