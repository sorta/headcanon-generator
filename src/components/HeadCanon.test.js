import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HeadCanon from './HeadCanon';

const sProps = {
  subject: {},
  descriptor: {},
  fandoms: { f_test_1: { name: 'Test Fandom 1' } },
};

// Tests for no subject or descriptor
describe("HeadCanon with no subject or descriptor", () => {

  it('renders and matches our snapshot', () => {
    const snapshot = renderer.create(
      <HeadCanon { ...sProps } />
    );
    const tree = snapshot.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('is null when no fandom is selected', () => {
    const component = shallow(<HeadCanon { ...sProps } />);
    expect(component.getElement()).toBe(null);
  });
});

const props = {
  ...sProps,
  descriptor: { name: 'Descriptor 1' },
  subject: { name: 'Subject 1', fandomKey: 'f_test_1' },
};

// Tests with subject and descriptor
describe("HeadCanon", () => {
  it('renders and matches our snapshot', () => {
    const snapshot = renderer.create(
      <HeadCanon { ...props } />
    );
    const tree = snapshot.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has a subject section', () => {
    const component = shallow(<HeadCanon { ...props } />);
    expect(component.find('.HeadCanon-Subject').exists()).toBe(true);
  });

  it('has a verb section', () => {
    const component = shallow(<HeadCanon { ...props } />);
    expect(component.find('.HeadCanon-Verb').exists()).toBe(true);
  });

  it('has a descriptor section', () => {
    const component = shallow(<HeadCanon { ...props } />);
    expect(component.find('.HeadCanon-Descriptor').exists()).toBe(true);
  });
});
