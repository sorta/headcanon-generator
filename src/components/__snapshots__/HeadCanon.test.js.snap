// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadCanon renders and matches our snapshot 1`] = `
<div
  className="HeadCanon"
>
  <h2
    className="HeadCanon-item HeadCanon-Subject"
    data-tooltip="From \\"Test Fandom 1\\""
  >
    Subject 1
  </h2>
   
  <h2
    className="HeadCanon-item HeadCanon-Verb"
  >
    is
  </h2>
   
  <h2
    className="HeadCanon-item HeadCanon-Descriptor"
  >
    Descriptor 1
  </h2>
</div>
`;

exports[`HeadCanon with no subject or descriptor renders and matches our snapshot 1`] = `null`;
